### PR TITLE
Avoid using publisher from OTM

### PR DESCRIPTION
**Pull request by jcague**
**Description**

There have been other crashes inside OneToManyProcessor when trying to access publisher's stuff.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

Not needed.

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: erizo/src/erizo/OneToManyProcessor.cpp [hunk 1]
```diff
@@ -69,13 +70,14 @@ namespace erizo {
       return 0;
     }
     boost::unique_lock<boost::mutex> lock(monitor_mutex_);
-    if (subscribers.empty())
+    if (subscribers_.empty() || !publisher_) {
```
Comment (lodoyun) [inline erizo/src/erizo/OneToManyProcessor.cpp]: 👍

File: erizo/src/erizo/OneToManyProcessor.h [hunk 2]
```diff
@@ -50,16 +52,20 @@ class OneToManyProcessor : public MediaSink, public FeedbackSink {
...
   int deliverAudioData_(std::shared_ptr<DataPacket> audio_packet) override;
   int deliverVideoData_(std::shared_ptr<DataPacket> video_packet) override;
   int deliverFeedback_(std::shared_ptr<DataPacket> fb_packet) override;
   int deliverEvent_(MediaEventPtr event) override;
   boost::future<void> closeAll();
   bool isSSRCFromAudio(uint32_t ssrc);
   uint32_t translateAndMaybeAdaptForSimulcast(uint32_t orig_ssrc);
+
+ private:
+  typedef std::shared_ptr<MediaSink> sink_ptr;
```
Comment (lodoyun) [inline erizo/src/erizo/OneToManyProcessor.h]: 💯 to the variable names updates. In that line, shouldn't this be `sink_ptr_` ?

File: erizoAPI/OneToManyProcessor.cc [hunk 3]
```diff
@@ -165,7 +165,7 @@ NAN_METHOD(OneToManyProcessor::getPublisherState) {
     return;
   }
 
-  auto wr = std::dynamic_pointer_cast<erizo::MediaStream>(me->publisher);
+  auto wr = std::dynamic_pointer_cast<erizo::MediaStream>(me->getPublisher());
```
Comment (lodoyun) [inline erizoAPI/OneToManyProcessor.cc]: 👍

File: erizo/src/erizo/OneToManyProcessor.h (hunk 2 above)
Comment (jcague) [reply erizo/src/erizo/OneToManyProcessor.h]: actually we can remove it because we don't use it

Comment (jcague) [reply erizo/src/erizo/OneToManyProcessor.h]: It was a typedef instead, but we don't use to define any variable in this class

**Pull request merged**